californium-core/src/main/java/org/eclipse/californium/core/network/CoapEndpoint.java

```diff
@@ -679,17 +679,20 @@ private void reject(final RawData raw, final MessageFormatException cause) {
 			rst.setToken(new byte[0]);
 			rst.setDestination(raw.getAddress());
 			rst.setDestinationPort(raw.getPort());
-			for (MessageInterceptor interceptor : interceptors) {
-				interceptor.sendEmptyMessage(rst);
-			}
-			connector.send(serializer.serializeEmptyMessage(rst));
+
+			reject(rst);
 			LOGGER.log(
 				Level.FINE,
 				"rejected malformed message from [{0}], reason: {1}",
 				new Object[]{raw.getInetSocketAddress(), cause.getMessage()});
 		}
 
 		private void reject(final Message message) {
+			if (!coapstack.supportsRejects()) {
+				LOGGER.log(Level.FINE, "Rejects not supported under this stack.");
+				return;
+			}
+
 			EmptyMessage rst = EmptyMessage.newRST(message);
```

Comment:
May be, I wrong, but on my first impression:

if the parameter message is a RST for a original message, the soruce/destination is already exchanged.

> EmptyMessage.newRST(message);

would exchange source/destination again.